src/cca/ui_components/server_list.ts

```diff
@@ -78,13 +80,14 @@ export class ServerListPage {
   }
 
   public pressAddServer(): Promise<ServerEntryComponent> {
-    return this.servers.addServer(this.addTokenText.value).then((server) => {
-      let entryElement = this.root.ownerDocument.createElement('div');
-      let serverEntry = new ServerEntryComponent(entryElement, server);
-      this.entryList.appendChild(entryElement);
-      return serverEntry;
-    }).catch((error) => {
-      console.error(error);
-    });
+    // TODO: wipe the list and re-restore if this throws
+    return this.servers.addServer(this.addTokenText.value);
+  }
+
+  // since this is used as a callback we must use the arrow notation.
+  private addServer = (server:Server) => {
```

Comment:
Problem is when the function is used as a callback...but I've changed the callers so this is no longer necessary.